chat-widget/src/firstresponselatency/FirstResponseLatencyTracker.ts

```diff
@@ -0,0 +1,156 @@
+import { LogLevel, TelemetryEvent } from "../common/telemetry/TelemetryConstants";
+import { MessagePayload, TrackingMessage } from "./Constants";
+
+import { TelemetryHelper } from "../common/telemetry/TelemetryHelper";
+
+export class FirstResponseLatencyTracker {
+
+    private isABotConversation = false;
+    private isStarted = false;
+    private isEnded = false;
+
+    private startTrackingMessage?: TrackingMessage;
+    private stopTrackingMessage?: TrackingMessage;
+
+    constructor() {
+        // this is a workaround to ensure in reload we track effectively the messages
+        // we do have a mechanism in place to prevent log agent messages.
+        this.isABotConversation = true;
+    }
+
+    private createTrackingMessage(payload: MessagePayload, type: string): TrackingMessage {
+        return {
+            Id: payload.Id,
+            role: payload.role,
+            timestamp: payload?.timestamp,
+            tags: payload.tags,
+            messageType: payload.messageType,
+            text: payload.text,
+            type: type,
+            checkTime: new Date().getTime()
+        };
+    }
+
+    // Tracking Functions
+    private startTracking(payload: MessagePayload): void {
+        //  this prevents to initiate tracking for multiple incoming messages
+        if (this.isStarted) {
+            return;
+        }
+        // this is to ensure we track only messages where bot is engaged
+        if (!this.isABotConversation) {
```

Comment:
Shall both `if` conditions be merged?

```
if (this.isStarted || !this.isABotConversation) return;
```

Reply:
This is one of those cases, where separation offers better readability